syntax = "proto3";
package mongo;




import "google/protobuf/struct.proto";

/*
Search请求
*/
message SearchRequest {
    /*
    表名
    */
    string collection = 1 ;
    /*
    查询条件
    */
    google.protobuf.Struct query = 2 ;
    /*
    fields
    */
    repeated string fields = 3 ;
    /*
    sort
    */
    repeated string sort = 4 ;
    /*
    页码
    */
    int32 page = 5 ;
    /*
    页大小
    */
    int32 pageSize = 6 ;
}

/*
Search返回
*/
message SearchResponse {
    /*
    总数
    */
    int32 total = 1 ;
    /*
    记录列表
    */
    repeated google.protobuf.Struct list = 2 ;
    /*
    页码
    */
    int32 page = 3 ;
    /*
    页大小
    */
    int32 pageSize = 4 ;
}

/*
SearchApi返回
*/
message SearchResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    mongo.SearchResponse data = 4 ;
}

